api/service/legacysync/helpers.go

```diff
@@ -58,39 +58,39 @@ func createSyncConfig(syncConfig *SyncConfig, peers []p2p.Peer, shardID uint32)
 	}
 	// limit the number of dns peers to connect
 	randSeed := time.Now().UnixNano()
-	peers = limitNumPeers(peers, randSeed)
+	targetSize, peers := limitNumPeers(peers, randSeed)
 
 	utils.Logger().Debug().
-		Int("len", len(peers)).
+		Int("peers count", len(peers)).
+		Int("target size", targetSize).
 		Uint32("shardID", shardID).
 		Msg("[SYNC] CreateSyncConfig: len of peers")
 
-	if len(peers) == 0 {
+	if targetSize == 0 {
 		return syncConfig, errors.New("[SYNC] no peers to connect to")
 	}
 	if syncConfig != nil {
 		syncConfig.CloseConnections()
 	}
 	syncConfig = NewSyncConfig(shardID, nil)
 
-	var wg sync.WaitGroup
+	var connectedPeers int
```

Comment:
What is the reason to remove async code?

Reply:
Because I need to count the connected peers one by one till they reach a certain number of connected peers. If I do parallel, first I have to wait for all client routines to finish and then disconnect the extra ones. Just imagine if we have a few hundred peers in our peer list and we need a few connected nodes. Does it make sense to connect to all of them (just to get the benefit of parallel routines) and then remove tenths of extra connections? And also, this function is called a few times, mostly at the beginning, and technically it won't affect performance that much, and in some certain scenarios it improves performance as well.